specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json

```diff
@@ -433,6 +433,35 @@
     },
     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/readonlykeys": {
       "get": {
+        "operationId": "DatabaseAccounts_GetReadOnlyKeys",
+        "x-ms-examples": {
+          "CosmosDBDatabaseAccountListReadOnlyKeys": { "$ref": "./examples/CosmosDBDatabaseAccountListReadOnlyKeys.json" }
+        },
+        "description": "Lists the read-only access keys for the specified Azure Cosmos DB database account.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/DatabaseAccountListReadOnlyKeysResult"
```

Comment:
It seems both `DatabaseAccounts_GetReadOnlyKeys` and `DatabaseAccounts_ListReadOnlyKeys` returns the same result. Are we planning to deprecate one of this eventually?

Reply:
Yes the GET is to be deprecated

Reply:
thank you. Is this new POST action to retrieve the keys a long running operation?

Reply:
No

Reply:
@dmakwana any reason why we use POST for listing keys?

Adding Gaurav - @ravbhatnagar  any concern here from ARM side? the context is:

1. There is already an operation to retrieve the database keys which is based on HTTP GET
2. We are introducing a new operation for retrieving the same database keys which is based on HTTP POST (this is not a long running operation)
3. The operation based on GET will be deprecated.

Reply:
@anuchandy - Yes, this is the right way. secrets like keys should not be exposed via  a GET. Right way is through a POST. Thanks for checking!

Reply:
Thanks @ravbhatnagar. 

@dmakwana merging the PR.